component/component/billing.jsonnet

```diff
@@ -0,0 +1,199 @@
+// main template for appuio-reporting
+local alerts = import 'billing_alerts.libsonnet';
+local common = import 'billing_cronjob.libsonnet';
+local netPol = import 'billing_netpol.libsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+// The hiera parameters for the component
+local params = inv.parameters.appcat;
+
+local formatImage = function(ref) '%(registry)s/%(repository)s:%(tag)s' % ref;
+
+local escape = function(str)
```

Comment:
A few comments to explain what this function does would be nice  (yes the name should be already a hint, but might still not be clear to everyone)

Reply:
I'm also not entirely sure, I've copied it from the appuio-reporting component.

But I think it just replaces any non-matching character with a `-`.